### PR TITLE
Implement regression tests using shell scripts under tests directory

### PR DESCRIPTION
**Pull request by binli2337**
* Implement regression testing using shell scripts.
* Regression tests using rt.sh passed all tests.
* Regression tests using NEMSCompsetRun passed all tests.
* Log files are updated.

### CODE REVIEW

**Comment by flampouris** [comment]
@binli2337 can you provide instructions on how to test it?

**Comment by binli2337** [comment]
To use rt.sh to do regression tests:

1. cd DATM-MOM6-CICE5/tests
2. rt.sh -f  (This will compile the code and run all 4 tests.)

Additional options will be provided on Wiki page after porting rt.sh to Orion is done.

**Comment by binli2337** [comment]
After tests are done successfully, the run directory will be removed automatically. To keep the run directory, please use "rt.sh -f -k".

**Comment by DeniseWorthen** [comment]
I'm a little surprised you didn't have trouble with the compile since I thought DATM didn't clean correctly (issue [#1](https://github.com/NOAA-EMC/NEMSdatm/issues/1)).

**Comment by flampouris** [comment]
I followed the running instructions and it runs without apparent error, but I cannot find the outputs, all the files at stmp4 have been deleted. 

@binli2337 could you please provide additional instructions on what I have to check and which log files I have to inspect?

(Maybe my questions are naive, but I am just a user that I have no experience with the RT).

**Comment by binli2337** [comment]
The log files are in tests directory.  The run directory is remove after a successful run!  Please see my comments earlier if you want to keep it.

File: parm/model_configure.IN [hunk 1]
```diff
@@ -17,6 +17,6 @@ atm_coupling_interval_sec:  @[coupling_interval_fast_sec]
 iatm: @[IATM]
 jatm: @[JATM]
 
-cdate0: @[CDATE]
+cdate0: @[SYEAR]@[SMONTH]@[SDAY]@[SHOUR]
```
Comment (flampouris) [inline parm/model_configure.IN]: This creates conflicts with downstream dependencies, GODAS.

**Review by flampouris — CHANGES_REQUESTED** [review]
@binli2337 thank you for preparing the RT.sh

I have several comments that are coming from the user's perspective and probably are trivial for everybody else.

1. It is unclear what these regression tests are testing. I can see the existence of the four regression tests: 
- tests/tests/cpld_datm_mom6_cice5_2d_cfsr 
- tests/tests/cpld_datm_mom6_cice5_2d_gefs 
- tests/tests/cpld_datm_mom6_cice5_cold_cfsr 
- tests/tests/cpld_datm_mom6_cice5_cold_gefs 

2. There are four log reports pushed, why?
- log/report-hera.intel-log/build_datm_mom6_cice.exe.log
- log/report-hera.intel-log/rtreport.txt
- tests/Compile_hera.intel.log
- tests/RegressionTests_hera.intel.log

Are these reports overwritten when the RTs are running?

3. The actual content of the reports seems to be the print out of the scripts with "set -x", but there is no actual confirmation message for (un)successful testing. 

4. The path structure is not very clear to me, the tests run at stmp2 but the new baseline is saved at stmp4. 

5. In the Compile_hera.intel.log, missing files and missing rules are indicated.

6. There are updates to the parm/*IN, see the comments in the code. 

7. I used the  "rt.sh -f -k" but at my run directory, I can see only the cpld_datm_mom6_cice5_cold_cfsr, see below.

Where are the other 3 tests?

Stylianos.Flampouris@hfe03 rt_193528]$ ls -rltha
total 24K
drwxr-sr-x 3 Stylianos.Flampouris stmp 4.0K May 20 14:34 ..
-rw-r--r-- 1 Stylianos.Flampouris stmp  744 May 20 14:42 run_test_001.env
drwxr-sr-x 3 Stylianos.Flampouris stmp 4.0K May 20 14:42 .
-rw-r--r-- 1 Stylianos.Flampouris stmp 6.7K May 20 14:42 cpld_datm_mom6_cice5_cold_cfsr.log
drwxr-sr-x 9 Stylianos.Flampouris stmp 4.0K May 20 14:42 cpld_datm_mom6_cice5_cold_cfsr

**Comment by DeniseWorthen** [comment]
@binli2337: I will review as soon as I get my RT tests launched for the CICE5 update.

**Comment by DeniseWorthen** [comment]
Stelios,
 
I just shared a presentation which may help explain the new system. I wasn't sure you had it or not. 

The Regression tests using rt.sh are just a replacement for the previous regression test system (NEMSCompsetRun) which was not portable to new platforms. So the four tests are just copies of the previous tests we had run with NEMSCompsetRun. 

I looked at your run (in /scratch1/NCEPDEV/stmp2/Stylianos.Flampouris/S2S_RT/rt_193528). If you look at the log file there, you'll see that the tests actually failed because of the ACCNR which is set for hera. When I run, I need to change this in rt.sh (LN 196) to marine-cpu; I don't know which account you need. 

I believe that for orion, the accnr is found automatically so you don't need to change it.

**Comment by binli2337** [comment]
The default account to use is "nems" and is set in detect_machine.sh.  If you cannot use "nems" to run tests, please set ACCNR in rt.sh.

**Review by flampouris — APPROVED** [review]
@binli2337 thank you for all the updates; it's an essential contribution.

I approve of the PR, the code can be merged, but I have a couple of additional comments, that it will be great if you could address them. 

1. It needs instructions on how to run the RT, explanation of the outputs, etc.

2. Some basic documentation, e.g., where the forcing files or the outputs are defined, etc.

3. The committed log files are confusing, and it does not make sense to me why they are part of the repository. If they are not required for the tests, could they be removed?

4. The structure of the regression test is different from the one described [here](https://docs.google.com/presentation/d/11xOmKCm2KIe6W4szvIGsFQaEbqNb6UtLmD-eYAflrZY/edit#slide=id.g8505e1abe5_0_15) are you considering to create something similar?

**Review by junwang-noaa — APPROVED** [review]

File: tests/datm_conf/cpld_datm_mom6_cice5_cfsr.IN [hunk 2]
```diff
@@ -0,0 +1,24 @@
+mkdir INPUT DATM_INPUT RESTART history MOM6_OUTPUT MOM6_RESTART restart
```
Comment (DeniseWorthen) [inline tests/datm_conf/cpld_datm_mom6_cice5_cfsr.IN]: I don't believe we are still using _MOM6_RESTART_ or _restart_; both MOM6 and CICE5 now write their restarts to the RESTART directory.

Comment (binli2337) [reply tests/datm_conf/cpld_datm_mom6_cice5_cfsr.IN]: These two directories will be removed.

**Review by DeniseWorthen — APPROVED** [review]

**Pull request merged**